source/validate_instruction.cpp

```diff
@@ -139,14 +154,33 @@ spv_result_t CapCheck(ValidationState_t& _,
     } else {
       // Check the operand word as a whole.
       const auto caps = RequiredCapabilities(_, operand.type, word);
-      if (!_.HasAnyOf(caps)) {
+      if (!_.HasAnyOfCapabilities(caps)) {
         return CapabilityError(_, i + 1, opcode, ToString(caps, _.grammar()));
       }
     }
   }
   return SPV_SUCCESS;
 }
 
+// Checks that all required extensions were declared in the module.
+spv_result_t ExtensionCheck(ValidationState_t& _,
+                            const spv_parsed_instruction_t* inst) {
+  const SpvOp opcode = static_cast<SpvOp>(inst->opcode);
```

Comment:
In practice so far, extensions are written so that only OpCapability has an extensions list.  Then other features are enabled by that capability.
But this is ok.